### PR TITLE
tls: drop tls.MinimumVersion in favor of tls.Options

### PR DESCRIPTION
**Pull request by hbagdi**
Drop tls.MinimumVersion as a defined property and instead use
tls.Options for defining tls.MinimumVersion.

### CODE REVIEW

**Comment by robscott** [comment]
Since `tls.options` is just a `map[string]string` it seems like it would be hard to standardize on common options like this. Are there other bits of common TLS config that would also end up in options?

File: apis/v1alpha1/generated.proto [hunk 1]
```diff
@@ -1196,18 +1196,6 @@ message TLSConfig {
...
-  // the TLS_* constants above. Note: MinimumVersion is not strongly
-  // typed to allow implementation-specific versions to be used without
-  // requiring updates to the API types. String must be of the form
-  // "<protocol><major>_<minor>".
-  //
-  // Support: Core for TLS1_{1,2,3}. Implementation-specific for all other
-  // values.
-  //
-  // +optional
-  optional string minimumVersion = 2;
```
Comment (yiyangy) [inline apis/v1alpha1/generated.proto:1209]: This changes the support of "minimumVersion" from core to implementation-specific. Is this intended?

**Comment by hbagdi** [comment]
To answer both @robscott and @yiyangy, for now, I think all TLS protocol specific will be folded inside `Options`. This indeed moves MinimumVersion from Core to ImplementationSpecific, we briefly discussed this in one of our meetings. We can introduce this back into core.

**Comment by bowei** [comment]
I'm OK with this, given the diversity of how TLS is specified.


/lgtm
/approve

**Pull request merged**